### PR TITLE
[experimental/stateHandling] Update to use cudaq::complex throughout 

### PR DESCRIPTION
**Pull request by amccaskey**
Per the language specification meeting - prefer `cudaq::complex` over `cudaq::simulation_scalar`, and introduce `cudaq::real`. 

Refactor `NoiseModel` to adhere to the simulation scalar type. 

Address #1539

### CODE REVIEW

File: runtime/common/NoiseModel.h [hunk 1]
```diff
@@ -63,12 +64,25 @@ struct kraus_op {
...
+  }
 
   /// @brief Return the adjoint of this kraus_op
-  kraus_op adjoint();
+  kraus_op adjoint() const {
+    std::size_t N = data.size();
+    std::vector<cudaq::complex> newData(N);
+    for (std::size_t i = 0; i < N; i++)
+      for (std::size_t j = 0; j < N; j++)
+        newData[j * N + i] = std::conj(data[i * N + j]);
```
Comment (1tnguyen) [inline runtime/common/NoiseModel.h:78]: Not sure if it's a bug but it seems like we're indexing beyond the length (`N`) of `newData`.
(perhaps the loop bounds are `nRows` and `nCols`?)

Comment (amccaskey) [reply runtime/common/NoiseModel.h:78]: My bad. Will fix.

File: runtime/cudaq/target_control.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+#pragma once
+
+namespace cudaq::__internal__ {
+
+/// @brief Provide an API call that enables
```
Comment (schweitzpgi) [inline runtime/cudaq/target_control.h:12]: Comments could be 80 column formatted in this file.

File: runtime/nvqir/custatevec/CuStateVecCircuitSimulator.cu [hunk 3]
```diff
@@ -335,7 +335,7 @@ protected:
         reinterpret_cast<CudaDataType *>(otherState),
         reinterpret_cast<CudaDataType *>(newDeviceStateVector));
     HANDLE_CUDA_ERROR(cudaGetLastError());
-
+
```
Comment (schweitzpgi) [inline runtime/nvqir/custatevec/CuStateVecCircuitSimulator.cu:338]: nit: adding spaces.

**Review by schweitzpgi — APPROVED** [review]

**Pull request merged**